api/tests/unit/app_analytics/test_unit_app_analytics_cache.py

```diff
@@ -0,0 +1,102 @@
+from app_analytics.cache import CACHE_FLUSH_INTERVAL, APIUsageCache
+from app_analytics.models import Resource
+from django.utils import timezone
+from freezegun import freeze_time
+from pytest_mock import MockerFixture
+
+
+def test_api_usage_cache(mocker: MockerFixture):
+    # Given
+    cache = APIUsageCache()
+    now = timezone.now()
+    mocked_track_request_task = mocker.patch("app_analytics.cache.track_request")
+    host = "host"
+    environment_key_1 = "environment_key_1"
+    environment_key_2 = "environment_key_2"
+
+    with freeze_time(now) as frozen_time:
+        # Make some tracking requests
+        for _ in range(10):
+            for resource in Resource:
+                cache.track_request(resource, host, environment_key_1)
+                cache.track_request(resource, host, environment_key_2)
+
+        # make sure track_request task was not called
+        assert not mocked_track_request_task.called
+
+        # Now, let's move the time forward
+        frozen_time.tick(CACHE_FLUSH_INTERVAL + 1)
+
+        # let's track another request(to trigger flush)
+        cache.track_request(
+            Resource.FLAGS,
+            host,
+            environment_key_1,
+        )
+
+        # Then - track request lambda was called for every resource and environment_key combination
+        expected_calls = []
+        for resource in Resource:
+            expected_calls.append(
+                mocker.call(
+                    kwargs={
+                        "resource": resource,
+                        "host": host,
+                        "environment_key": environment_key_1,
+                        "count": 11 if resource == Resource.FLAGS else 10,
+                    }
+                )
+            )
+            expected_calls.append(
+                mocker.call(
+                    kwargs={
+                        "resource": resource,
+                        "host": host,
+                        "environment_key": environment_key_2,
+                        "count": 10,
+                    }
+                )
+            )
+        mocked_track_request_task.delay.assert_has_calls(expected_calls)
+
+
+def test_api_usage_cache__flush_resets_cache_and_interval(
+    mocker: MockerFixture,
+):
+    # Given
+    mocked_track_request_task = mocker.patch("app_analytics.cache.track_request")
+    cache = APIUsageCache()
+    now = timezone.now()
+
+    with freeze_time(now) as frozen_time:
+        # Add some data
+        cache.track_request(Resource.FLAGS, "host", "environment_key")
+
+        # move the time forward
+        frozen_time.tick(CACHE_FLUSH_INTERVAL + 1)
+
+        # Next, let's call flush
+        cache.flush()
```

Comment:
I think we have this pattern elsewhere so we're probably just following that, but it's odd to me that we're calling `flush` directly here since I don't think any external calls are made to it. I think we should probably mark the `flush` method as private, and this test should only use the `track_request` method.

Reply:
Although It's not being used, but I feel it makes the interface a bit more clean? Does that make sense?

Reply:
Hmm... maybe it does, yes, but the test feels a bit too 'internal' to me. It's not really testing the actual behaviour.

Reply:
Yeah, that's a fair point! I have made it private